tests/test_page.py

```diff
@@ -10,23 +10,48 @@
 
 
 def test_base_url(base_url, page):
-    assert base_url == page.url
+    assert base_url == page.canonical_url
 
 
-def test_url_keywords(base_url, selenium):
+def test_canonical_url_absolute(base_url, selenium):
+    url_template = 'https://www.test.com/'
+
+    class MyPage(Page):
+        URL_TEMPLATE = url_template
+    page = MyPage(selenium, base_url)
+    assert url_template == page.canonical_url
+
+
+def test_canonical_url_empty(selenium):
+    page = Page(selenium)
+    assert page.canonical_url == ''
```

Comment:
Am I being overly picky? Perhaps `URL_TEMPLATE` could default to `None` and then in `canonical_url` we could do:

```python
if self.URL_TEMPLATE is not None:
    return urljoin(self.base_url, self.URL_TEMPLATE.format(**self.url_kwargs))
return base_url
```